src/shared/components/plots/WaterfallPlot.spec.tsx

```diff
@@ -0,0 +1,182 @@
+            datum1 = plot.internalData[0];
+            datum2 = plot.internalData[1];
+            assert.equal(datum1.order, 1);
+            assert.equal(datum1.value, 4);
+            assert.equal(datum2.order, 2);
+            assert.equal(datum2.value, 2);
+        });
+        
+    });
+    
+    describe('Plot domain', () => {
+        it('Has correct x dimensions', () => {
+            const domain = plot.plotDomain;
+            assert.equal(domain.order[0], 1);
+            assert.equal(domain.order[1], 2);
+        });
+        it('Has correct y dimensions', () => {
+            const domain = plot.plotDomain;
+            assert.equal(domain.value[0], 2);
+            assert.equal(domain.value[1], 4);
+        });
+    });
+
+    describe('Search indicators', () => {
+
+        it('Absent when no highlight param passed', () => {
+            testProps.highlight = undefined;
+            assert.equal(plot.searchLabels.length, 0);
+        });
+        
+        it('Absent when no samples match', () => {
+            testProps.highlight = (d) => false;
+            assert.equal(plot.searchLabels.length, 0);
+        });
+        
+        it('Present when search function is passed', () => {
+            testProps.highlight = (d) => true;
+            assert.equal(plot.searchLabels.length, 2);
+        });
+
+        it('Present when search function is passed', () => {
+            testProps.highlight = (d) => true;
+            assert.equal(plot.searchLabels.length, 2);
+        });
+
+        it('Search label data points have x/y coordinates', () => {
+            testProps.highlight = (d) => true;
+            assert.isDefined(plot.searchLabels[0].searchindicatorx);
+            assert.isDefined(plot.searchLabels[0].searchindicatory);
+        });
+
+    });
+
+    describe('Limit value indicators', () => {
+
+        it('Absent when labelVisibility param not passed', () => {
+            testProps.labelVisibility = undefined;
+            plot = new WaterfallPlot<IBaseWaterfallPlotData>(testProps);
+            assert.equal(plot.limitValueLabels.length, 0);
+        });
+
+        it('Absent when no datapoints are limited', () => {
+           testProps.labelVisibility = () => false;
+           plot = new WaterfallPlot<IBaseWaterfallPlotData>(testProps);
+           assert.equal(plot.limitValueLabels.length, 0);
+        });
+
+        it('Present when datapoints are limited', () => {
+            testProps.labelVisibility = () => true;
+            plot = new WaterfallPlot<IBaseWaterfallPlotData>(testProps);
+            assert.equal(plot.limitValueLabels.length, 2);
+        });
+
+        it('Limit vlaue label data points have x/y coordinates', () => {
+            testProps.labelVisibility = () => true;
+            plot = new WaterfallPlot<IBaseWaterfallPlotData>(testProps);
+            assert.isDefined(plot.limitValueLabels[0].labelx);
+            assert.isDefined(plot.limitValueLabels[0].labely);
+        });
+        
```

Comment:
overall these are great tests

Reply:
@alisman what do you think of this approach to testing (checking public properties on the instantiated component)

Reply:
I used this approach since this component is just a layer between Plotstab and Victory chart. I just test what data this component passes onto Victory. I realize this is very implementation dependent. But I saw no other way to test the functionality of WaterfallPlot.tsx. Feedback would be much appreciated.

Reply:
the other option would be to move the logic to a helper function. In a way I prefer that since it makes the test not dependent on more arbitrary types of implementation details. But in this case it doesn't bother me and I see the benefits of doing it like this because it's almost like an end to end test. I'm cool with it.